File: instrumentation/startup/src/main/java/io/opentelemetry/android/instrumentation/startup/InitializationEvents.java
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */

package io.opentelemetry.android.instrumentation.startup;

import io.opentelemetry.sdk.trace.export.SpanExporter;
import java.util.Map;

public interface InitializationEvents {

    void sdkInitializationStarted();

    void recordConfiguration(Map<String, String> config);

    void currentNetworkProviderInitialized();

    void networkMonitorInitialized();

    void anrMonitorInitialized();

    void slowRenderingDetectorInitialized();

    void crashReportingInitialized();

    void spanExporterInitialized(SpanExporter spanExporter);

    InitializationEvents NO_OP =
            new InitializationEvents() {
                @Override
                public void sdkInitializationStarted() {}

                @Override
                public void recordConfiguration(Map<String, String> config) {}

                @Override
                public void currentNetworkProviderInitialized() {}

                @Override
                public void networkMonitorInitialized() {}

                @Override
                public void anrMonitorInitialized() {}

                @Override
                public void slowRenderingDetectorInitialized() {}

                @Override
                public void crashReportingInitialized() {}

                @Override
                public void spanExporterInitialized(SpanExporter spanExporter) {}
            };
}

File: instrumentation/startup/src/main/java/io/opentelemetry/android/instrumentation/startup/InitializationEvents.kt
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */

package io.opentelemetry.android.instrumentation.startup

import io.opentelemetry.sdk.trace.export.SpanExporter

interface InitializationEvents {
    fun sdkInitializationStarted()

    fun recordConfiguration(config: Map<String, String>)

    fun currentNetworkProviderInitialized()

    fun networkMonitorInitialized()

    fun anrMonitorInitialized()

    fun slowRenderingDetectorInitialized()

    fun crashReportingInitialized()

    fun spanExporterInitialized(spanExporter: SpanExporter)

    companion object {
        @JvmField
        val NO_OP: InitializationEvents =
            object : InitializationEvents {
                override fun sdkInitializationStarted() {}

                override fun recordConfiguration(config: Map<String, String>) {}

                override fun currentNetworkProviderInitialized() {}

                override fun networkMonitorInitialized() {}

                override fun anrMonitorInitialized() {}

                override fun slowRenderingDetectorInitialized() {}

                override fun crashReportingInitialized() {}

                override fun spanExporterInitialized(spanExporter: SpanExporter) {}
            }
    }
}
